cardano-node/app/chairman.hs

```diff
@@ -21,8 +20,42 @@ import           Ouroboros.Consensus.Node.ProtocolInfo.Abstract (NumCoreNodes (.
 import           Ouroboros.Consensus.NodeId (CoreNodeId)
 
 import           Cardano.Node.CLI
+import           Cardano.Node.Parsers (parseCoreNodeId, parseProtocol)
 
-import           Chairman (runChairman)
+import           Cardano.Chairman (runChairman)
+
+main :: IO ()
+main = do
+    ChairmanArgs { caProtocol
+                 , caCoreNodeIds
+                 , caSecurityParam
+                 , caMaxBlockNo
+                 , caTimeout
+                 , caCommonCLI
+                 } <- execParser opts
+
+    SomeProtocol p
+      <- case finaliseCardanoConfiguration $
+                mergeConfiguration
+                  mainnetConfiguration
+                  caCommonCLI of
+        Left err -> throwIO (ConfigurationError err)
+        Right cc -> fromProtocol cc caProtocol
+
+    let run = runChairman p caCoreNodeIds
+                          (NumCoreNodes $ length caCoreNodeIds)
+                          caSecurityParam
+                          caMaxBlockNo
+                          stdoutTracer
+
+    case caTimeout of
+      Nothing      -> run
+      Just timeout ->
+        run
+        `race_`
+        do
+          threadDelay (timeout * 1_000_000)
```

Comment:
The `1_000_000` confused me initially but apparently this is something introduced by the `NumericUnderscores` extension.